snuba/datasets/processors/__init__.py

```diff
@@ -0,0 +1,53 @@
+from __future__ import annotations
+
+import os
+from abc import abstractmethod
+from typing import Any, Optional, Type, cast
+
+from snuba.consumers.types import KafkaMessageMetadata
+from snuba.processor import MessageProcessor, ProcessedMessage
+from snuba.utils.registered_class import (
+    RegisteredClass,
+    camel_to_snake,
+    import_submodules_in_directory,
+)
+
+
+class MessageProcessorUnsupportedFromConfig(Exception):
+    pass
+
+
+class DatasetMessageProcessor(MessageProcessor, metaclass=RegisteredClass):
+    """
+    The Processor is responsible for converting an incoming message body from the
+    event stream into a row or statement to be inserted or executed against clickhouse.
+
+    The DatasetMessageProcessor is responsible for the message processors that write data
+    for a storage.
+    """
+
+    def __init__(self) -> None:
+        pass
+
+    @classmethod
+    def get_from_name(cls, name: str) -> Type["DatasetMessageProcessor"]:
+        return cast(Type["DatasetMessageProcessor"], cls.class_from_name(name))
+
+    @classmethod
+    def from_kwargs(cls, **kwargs: str) -> DatasetMessageProcessor:
+        return cls(**kwargs)
+
+    @classmethod
+    def config_key(cls) -> str:
+        return camel_to_snake(cls.__name__)
```

Comment:
I am not sure how I feel about this. If we specified the class name exactly in the configuration, we could get rid of a lot of logic. It does ruin the aesthetics of the configuration though.